src/web/components/SignInGate/displayRule.ts

```diff
@@ -79,15 +83,53 @@ export const canShow = (
 	CAPI: CAPIBrowserType,
 	isSignedIn: boolean,
 	currentTest: CurrentABTest,
-): boolean =>
-	!isSignedIn &&
-	!hasUserDismissedGateMoreThanCount(
-		currentTest.variant,
-		currentTest.name,
-		5,
-	) &&
-	isNPageOrHigherPageView(3) &&
-	isValidContentType(CAPI) &&
-	isValidSection(CAPI) &&
-	isValidTag(CAPI) &&
-	!isIOS9();
+): Promise<boolean> => {
+	return Promise.resolve(
+		!isSignedIn &&
+			!hasUserDismissedGateMoreThanCount(
+				currentTest.variant,
+				currentTest.name,
+				5,
+			) &&
+			isNPageOrHigherPageView(3) &&
+			isValidContentType(CAPI) &&
+			isValidSection(CAPI) &&
+			isValidTag(CAPI) &&
+			!isIOS9(),
+	);
+};
+
+export const hasRequiredConsents = (): Promise<boolean> => {
+	const consentsRequiredFor: ['google-analytics', 'ophan', 'sentry'] = [
```

Comment:
we need to confirm whether these are all the consents required for mandatory signin

Reply:
We need to check for "accept all" consents as said in standup.
It would be useful if the CMP returned a list of id's to check, or a type that returns all the id's.
This will get real messy if we have to manually add all the ids.

Maybe one for @guardian/commercial-dev can help us with/advise?

Reply:
I've implemented a temporary solution https://github.com/guardian/dotcom-rendering/pull/2661/commits/4c1fb8853313f08114ff1f9bc643a5446eecc1d0 , but it's not ideal